astroquery/alma/core.py

```diff
@@ -604,7 +604,7 @@ def get_data_info(self, uids, *, expand_tarfiles=False,
         if expand_tarfiles:
             for index, row in enumerate(result):
                 # Recursive DataLink, so look for service_def
-                if row['service_def'] and row['content_type'] == DATALINK_FILE_TYPE:
+                if row['service_def']:
```

Comment:
Yes, the current code is simply incorrect, since the `content_type` returned by a tarball is not the `DATALINK_FILE_TYPE` presently implemented.  If the solution is to change `DATALINK_FILE_TYPE`, please propose that change!

Reply:
Apparently `DATALINK_FILE_TYPE` was totally removed since I made this PR.

Reply:
I think @at88mph has re-worked this part in the meantime, so the change and my comment are outdated.